pact-jvm-consumer-kotlin/src/test/kotlin/au/com/dius/pact/consumer/kotlin/dsl/examples/DesignDecisions.kt

```diff
@@ -0,0 +1,148 @@
+                    whenever receiving "a request for something" withPath "/hello" and {
+                        method("POST")
+                        body("{\"name\": \"harry\"}")
+                    } then respond with {
+                        status(404)
+                        body("{\"hello\": \"harry\"}")
+                    }
+
+                    whenever receiving "another request for something" withPath "/hello" and {
+                        method("POST")
+                        body("{\"name\": \"harry\"}")
+                    } then respond with {
+                        status(404)
+                        body("{\"hello\": \"harry\"}")
+                    }
+                }
+            }
+
+        Assertions.assertThat(infixDsl).isEqualTo(ConsumerPactBuilderExample.consumerPactBuilderExample)
+    }
+
+    @Test
+    fun proseLikeShorter() {
+        val infixDsl =
+        // tag::proseLikeShorter[]
+            KPact between "Some Consumer" and "Some Provider" { // <1> <2>
+                given providerState "a certain state on the provider" { // <3>
+                    whenever receiving "a request for something" withPath "/hello" { // <4>
+                        method("POST")
+                        body("{\"name\": \"harry\"}")
+                    } then respond with { // <5>
+                        status(200)
+                        body("{\"hello\": \"harry\"}")
+                    }
+                    // end::proseLikeShorter[]
+
+                    whenever receiving "another request for something" withPath "/hello" {
+                        method("POST")
+                        body("{\"name\": \"harry\"}")
+                    } then respond with {
+                        status(200)
+                        body("{\"hello\": \"harry\"}")
+                    }
+                }
+
+                given providerState "other state on the provider" {
+                    whenever receiving "a request for something" withPath "/hello" {
+                        method("POST")
+                        body("{\"name\": \"harry\"}")
+                    } then respond with {
+                        status(404)
+                        body("{\"hello\": \"harry\"}")
+                    }
+
+                    whenever receiving "another request for something" withPath "/hello" {
+                        method("POST")
+                        body("{\"name\": \"harry\"}")
+                    } then respond with {
+                        status(404)
+                        body("{\"hello\": \"harry\"}")
+                    }
+                }
+            }
+
+        Assertions.assertThat(infixDsl).isEqualTo(ConsumerPactBuilderExample.consumerPactBuilderExample)
+    }
+
+    @Test
+    fun infixDsl() {
+        val infixDsl =
+        // tag::CamelCase[]
+            KPact between "Some Consumer" and "Some Provider" isDefinedBy { // <1> <2>
```

Comment:
I prefer this one. One word usually makes it easier to use in an IDE. I personally like it to be a little more explicit, thus I prefer this over the very short option.